### ISSUE TITLE
Pathfinding dies with invalid path

### ISSUE THREAD

**Issue by JoshuaJSam** [UBCSailbot/local-pathfinding#215]
![image](https://user-images.githubusercontent.com/47064054/125174402-218bc580-e17a-11eb-9cc4-74baee8d9e4a.png)

**Comment by tylerlum** [comment]
Reproduce with:
```
➜  local-pathfinding git:(master) ✗ roslaunch local_pathfinding all.launch num_ais_ships:=2000 random_seed:=4100 runtime_seconds:=1.0 num_runs:=1 speedup:=200
```

![image](https://user-images.githubusercontent.com/26510814/125178730-56107900-e19c-11eb-902f-33346ca3696a.png)

**Comment by tylerlum** [comment]
Resolved in #219
